test/spec/Feature/Query/AggregateFunctionsSpec.hs

```diff
@@ -0,0 +1,70 @@
+          [json|[{ "min": 1 }]|] { matchHeaders = [matchContentTypeJson] }
+      it "supports max()" $
+        get "/car_model_sales?select=quantity.max()" `shouldRespondWith`
+          [json|[{ "max": 9 }]|] { matchHeaders = [matchContentTypeJson] }
+      it "supports count()" $
+        get "/car_model_sales?select=car_model_name.count()" `shouldRespondWith`
+          [json|[{ "count": 4 }]|] { matchHeaders = [matchContentTypeJson] }
+      it "groups by any fields selected that do not have an aggregate applied" $
+        get "/car_model_sales?select=quantity.sum(),quantity.max(),date.min(),car_model_name" `shouldRespondWith`
+          [json|[{ "sum": 4, "max": 3, "min": "2021-02-11", "car_model_name": "Murcielago"}, { "sum": 16, "max": 9, "min": "2021-01-14", "car_model_name": "DeLorean"}]|] { matchHeaders = [matchContentTypeJson] }
+      it "supports the use of aliases on fields that will be used in the group by" $
+        get "/car_model_sales?select=quantity.sum(),quantity.max(),date.min(),cm:car_model_name" `shouldRespondWith`
+          [json|[{ "sum": 4, "max": 3, "min": "2021-02-11", "cm": "Murcielago"}, { "sum": 16, "max": 9, "min": "2021-01-14", "cm": "DeLorean"}]|] { matchHeaders = [matchContentTypeJson] }
+      it "allows you to specify an alias for the aggregate" $
+        get "/car_model_sales?select=total_sold:quantity.sum(),car_model_name" `shouldRespondWith`
+          [json|[{ "total_sold": 4, "car_model_name": "Murcielago"}, { "total_sold": 16, "car_model_name": "DeLorean" }]|] { matchHeaders = [matchContentTypeJson] }
+      it "allows you to cast the result of the aggregate" $
+        get "/car_model_sales?select=total_sold:quantity.sum()::text,car_model_name" `shouldRespondWith`
+          [json|[{ "total_sold": "4", "car_model_name": "Murcielago"}, { "total_sold": "16", "car_model_name": "DeLorean" }]|] { matchHeaders = [matchContentTypeJson] }
+      it "allows you to cast the input argument of the aggregate" $
+        get "/trash_details?select=jsonb_col->>key::integer.sum()" `shouldRespondWith`
+          [json|[{"sum": 24}]|] { matchHeaders = [matchContentTypeJson] }
+      it "allows the combination of an alias, a before cast, and an after cast" $
+        get "/trash_details?select=s:jsonb_col->>key::integer.sum()::text" `shouldRespondWith`
+          [json|[{"s": "24"}]|] { matchHeaders = [matchContentTypeJson] }
+      it "supports use of aggregates on RPC functions that return table values" $
+        get "/rpc/getallprojects?select=id.max()" `shouldRespondWith`
+          [json|[{"max": 5}]|] { matchHeaders = [matchContentTypeJson] }
+    context "performing aggregations on spreaded fields from an embedded resource" $ do
+      xit "supports the use of aggregates on spreaded fields" $
+        get "/car_models?select=car_brand_name,...car_model_sales(quantity.sum())" `shouldRespondWith`
+          [json|[{"car_brand_name": "dog"}]|] { matchHeaders = [matchContentTypeJson] }
```

Comment:
Since this one is marked as [pending](https://hackage.haskell.org/package/hspec-2.11.7/docs/Test-Hspec.html#v:xit), is that intentional?

Reply:
Hoisting works, but the test here is broken because this relationship doesn't actually support spreads at all, with or without aggregates.

Reply:
e.g.

```curl --location 'http://localhost:3000/contract?select=tournament%2C...player_view(birth_date.min())'```

Reply:
I've added tests for the hoisting.